pkg/cmd/historical_logs.go

```diff
@@ -0,0 +1,149 @@
+package cmd
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"github.com/spf13/cobra"
+	"io/ioutil"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/kubectl/pkg/util/i18n"
+	"k8s.io/kubectl/pkg/util/templates"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+var
+(
+	logsExample = templates.Examples(i18n.T(`
+		# Return snapshot logs from pod openshift-apiserver-operator-849d7869ff-r94g8 with a maximum of 10 Entries
+		kubectl historical-logs --podname=openshift-apiserver-operator-849d7869ff-r94g8 --maxlogs=10
+		# Return snapshot logs from namespace openshift-apiserver-operator and logging level info
+		kubectl historical-logs --namespace=openshift-apiserver-operator --level=info
+		# Return snapshot logs from every index without filtering
+		kubectl historical-logs
+		# Return snapshot of historical-logs from Infrastructure index
+		kubectl historical-logs --index=infra-000001
+		# Return snapshot logs in a time range between start and end times
+		kubectl historical-logs --starttime=2021-03-09T03:40:00.163677339Z --finishtime=2021-03-09T03:50:00.163677339Z
+		# Return snapshot logs in a time range between start and end times for the infrastructure index
+		kubectl historical-logs --starttime=2021-03-09T03:40:00.163677339Z --finishtime=2021-03-09T03:50:00.163677339Z --index=infra-000001
+		# Return snapshot a maximum of 20 historical-logs`))
+)
+
+type LogParameters struct {
+	Namespace string `json:"namespace"`
+	Index string `json:"index"`
+	Podname string `json:"podname"`
+	StartTime string `json:"starttime"`
+	FinishTime string `json:"finishtime"`
+	Level string `json:"level"`
+	MaxLogs string `json:"maxlogs"`
+}
+
+type ResponseLogs struct{
+	Logs []string `json:"Logs"`
+}
+
+
+func NewCmdLogFilter(streams genericclioptions.IOStreams) *cobra.Command {
+
+	o := &LogParameters{}
+	cmd := &cobra.Command{
+		Use:     "historical-logs [flags]",
+		Short:   "View logs filtered on various parameters",
+		Example: logsExample,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			err := o.Execute(streams)
+			if err!=nil {
+				return err
+			}
+			return nil
+		},
+	}
+
+	o.AddFlags(cmd)
+	return cmd
+}
+
+func (o *LogParameters)AddFlags(cmd *cobra.Command) {
+
+	cmd.Flags().StringVar(&o.Podname, "podname", "", "Filter Historical logs on a specific pod name.")
```

Comment:
Allow users to specify different resource types like deployment, daemonset, etc.